### PR TITLE
Add COS integration

### PR DESCRIPTION
**Pull request by weiiwang01**
### Overview

Add COS integration (loki, prometheus, grafana) to the Penpot charm.

Screenshot 2024-10-14 at 00 49 25

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: src/grafana_dashboards/penpot.json [hunk 1]
```diff
@@ -0,0 +1,753 @@
...
+        "x": 0,
+        "y": 8
+      },
+      "id": 2,
+      "options": {
+        "legend": {
+          "calcs": [],
+          "displayMode": "list",
+          "placement": "bottom",
+          "showLegend": true
```
Comment (cbartz) [inline src/grafana_dashboards/penpot.json:356]: Could we refer to the juju unit or instance in the legend? The screenshot shows that all labels are included in the legend, which may not be very legible.

File: src/grafana_dashboards/penpot.json
Comment (cbartz) [inline src/grafana_dashboards/penpot.json:1]: Could you use seconds for the units on the y axes?

**Review by cbartz — COMMENTED** [review]
Thanks. Some comments on the dashboard.

File: src/grafana_dashboards/penpot.json [hunk 2]
```diff
@@ -0,0 +1,753 @@
...
+  },
+  "time": {
+    "from": "now-3h",
+    "to": "now"
+  },
+  "timepicker": {},
+  "timezone": "",
+  "title": "Penpot Dashboard",
+  "uid": null,
+  "version": 3,
```
Comment (amandahla) [inline src/grafana_dashboards/penpot.json:751]: version 1?

File: src/charm.py [hunk 3]
```diff
@@ -47,6 +50,17 @@ def __init__(self, *args: typing.Any):
...
+        self._metrics_endpoint = MetricsEndpointProvider(
+            self,
+            jobs=[
+                {
+                    "job_name": "penpot_metrics",
+                    "static_configs": [{"targets": ["*:6060"]}],
+                }
+            ],
+        )
+        self._logg_forwarder = LogForwarder(self)
```
Comment (amandahla) [inline src/charm.py:63]: minor/nit: _log_forwarder

File: src/grafana_dashboards/penpot.json (hunk 1 above)
Comment (weiiwang01) [reply src/grafana_dashboards/penpot.json:356]: Yes, updated, thanks!

File: src/grafana_dashboards/penpot.json
Comment (weiiwang01) [reply src/grafana_dashboards/penpot.json:1]: Changed the unit for the Y axes, thanks!

File: src/grafana_dashboards/penpot.json (hunk 2 above)
Comment (weiiwang01) [reply src/grafana_dashboards/penpot.json]: Updated, thanks!

File: src/charm.py (hunk 3 above)
Comment (weiiwang01) [reply src/charm.py]: Updated, thanks!

**Review by jdkandersson — DISMISSED** [review]

**Review by cbartz — DISMISSED** [review]

**Review by jdkandersson — APPROVED** [review]

File: .github/workflows/test.yaml [hunk 4]
```diff
@@ -18,7 +18,5 @@ jobs:
       channel: 1.28-strict/stable
       charmcraft-channel: latest/edge
       juju-channel: 3.4/stable
-      self-hosted-runner: true
-      self-hosted-runner-label: "edge"
```
Comment (cbartz) [inline .github/workflows/test.yaml:22]: Why are you removing this?

Comment (weiiwang01) [reply .github/workflows/test.yaml:22]: The integration tests failed to run in the self-hosted runner environment, and I can't reproduce the error locally. I will investigate this problem later, but for now, just switch to using the GitHub-hosted runners to merge this pull request.

Comment (cbartz) [reply .github/workflows/test.yaml:22]: Ok, there were some infrastructure issues. I think the issues could be resolved. Please re-enable in the next PR.

**Review by cbartz — APPROVED** [review]

**Pull request merged**